magentov2/plugin/Indodana/PayLater/Helper/Transaction.php

```diff
@@ -0,0 +1,514 @@
+  public function __construct(
+    Data $helper,
+    \Magento\Customer\Api\CustomerRepositoryInterface $customerRepositoryInterface,
+    \Magento\Framework\UrlInterface $urlInterface,
+    \Magento\Framework\App\Filesystem\DirectoryList $directoryList
+    )
+  {
+          $this->_helper = $helper;
+          $this->_customer= $customerRepositoryInterface;
+          $this->_urlInterface = $urlInterface;
+          $this->_dir = $directoryList;
+          require_once($this->_dir->getPath(DirectoryList::APP). '/code/Indodana/PayLater/autoload.php' );
+  }
+
+  public function getIndodanaCommon()
+  {
+    if (!isset($this->indodanaCommon)) {
+      $apiKey = $this->_helper->getApiKey();
+      $apiSecret = $this->_helper->getApiSecret();
+      $environment = $this->_helper->getEnvironment();
+
+      $this->indodanaCommon = new IndodanaCommon([
+        'apiKey'        => $apiKey,
+        'apiSecret'     => $apiSecret,
+        'environment'   => $environment,
+        'seller'        => $this->getSeller()
+      ]);
+    }
+    return $this->indodanaCommon;
+  }
+
+  public function getTotalAmount($order)
+  {
+    return (float) $order->getGrandTotal();
+  }
+
+  public function getTotalDiscountAmount($order)
+  {
+    $totalDiscountAmount = 0;
+    foreach ($order->getAllVisibleItems() as $item) {
+      $totalDiscountAmount += (float) $item->getDiscountAmount();
+    }
+    return $totalDiscountAmount;
+  }
+
+  public function getTotalShippingAmount($order)
+  {
+    // I'm not really sure whether to use getShippingInclTax() or getShippingAmount()
+    // For get installment options
+    $totalShippingAmount = (float) $order->getShippingAddress()->getShippingInclTax();
+    // For checkout
+    if (!$totalShippingAmount) {
+      $totalShippingAmount = (float) $order->getShippingInclTax();
+    }
+    return $totalShippingAmount;
+  }
+
+  public function getTotalTaxAmount($order)
+  {
+    $totalTaxAmount = 0;
+    foreach ($order->getAllVisibleItems() as $item) {
+      $totalTaxAmount += (float) $item->getTaxAmount();
+    }
+    return $totalTaxAmount;
+  }
+
+  public function getProducts($order)
+  {
+    $objectManager = \Magento\Framework\App\ObjectManager::getInstance();
+    // We only need parent products
+    $orderItems = $order->getAllVisibleItems();
+    $products = [];
+    foreach($orderItems as $orderItem) {
+      $product = $orderItem->getProduct();
+      // For get installment options
+      $quantity = $orderItem->getQty();
+      // For checkout
+      if (!$quantity) {
+        $quantity = $orderItem->getQtyToInvoice();
+      }
+      $cat = $objectManager->create('Magento\Catalog\Model\Category')->load($product->getCategoryIds()[0]);
+      $products[] = [
+        'id'        => $product->getId(),
+        'name'      => $product->getName(),
+        'price'     => (float) $product->getPrice(),
+        'url'       => $product->getProductUrl(),
+        'imageUrl'  => '', // TODO: Search how to do this
```

Comment:
What is the value for this in Magento V1?

Reply:
$imageUrl = $this->imageHelperFactory->create()
        ->init($product, 'product_thumbnail_image')->getUrl();      

      $products[] = [
        'id'        => $product->getId(),
        'name'      => $product->getName(),
        'price'     => (float) $product->getPrice(),
        'url'       => $product->getProductUrl(),
        'imageUrl'  => $imageUrl, 
        'type'      => $product->getTypeId(), 
        'quantity'  => $quantity,
        'category' =>  IndodanaConstant::DEFAULT_ITEM_CATEGORY,
        'parentId' => $this->_helper->getStoreID()
      ];